promql/parser/parse_test.go

```diff
@@ -1638,14 +1668,6 @@ var testExpr = []struct {
 		input:  `foo[0m]`,
 		fail:   true,
 		errMsg: "duration must be greater than 0",
-	}, {
-		input:  `foo[5m30s]`,
```

Comment:
Some new negative tests would be good, like 1hm, 1m1, 1m2h, 1m1m.

Reply:
added negative tests